src/CMakeLists.txt

```diff
@@ -377,4 +378,28 @@ install (TARGETS ${PROJECT_NAME}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
-install (EXPORT ${PROJECT_NAME}-target DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME} FILE ${PROJECT_NAME}-config.cmake)
+
+# Generate and install CMake package
+include(CMakePackageConfigHelpers)
+set(PACKAGE_INSTALL_DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}-${NN_PACKAGE_VERSION}
+)
+install(EXPORT ${PROJECT_NAME}-target
+    DESTINATION ${PACKAGE_INSTALL_DESTINATION}
+)
+write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake
+    VERSION ${NN_PACKAGE_VERSION}
+    COMPATIBILITY AnyNewerVersion
+)
+configure_package_config_file(
+    ${CMAKE_SOURCE_DIR}/cmake/${PROJECT_NAME}-config.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake
+    INSTALL_DESTINATION ${PACKAGE_INSTALL_DESTINATION}
+    PATH_VARS CMAKE_INSTALL_PREFIX
+)
+install(FILES
```

Comment:
This no longer has an EXPORT associated with it.  Is that correct?  I'm still trying to come to grips with the relationships within CMake here.

Reply:
Yes, the `EXPORT` property is only needed when installing targets for which an export set should be generated and installed (a file written in cmake that lists all exported targets and is meant to be imported by the consumer).

Next to the export sets, you can have these two special _top-level_ files in your CMake package. `*-config.cmake` is meant to export all sorts of information that might be useful to your user. `*-config-version.cmake` implements how  the user-supplied version (e.g. `find_package(nanomsg 1.1.4)`) is evaluated. In this PR I use the standard implementation shipped with CMake (it relies on the CMake built-in operator `VERSION_GREATER|EQUAL|LESS`). But if you wanted to implement a custom order on your versions, you could replace the call to `write_basic_package_version_file(...)` with your own file.

I tested that the default implementation from CMake works with all possible `NN_PACKAGE_VERSION`s, They can have the format `1.1.4` or `1.1.4-3-g45h5kj5` or `1.1.4-3-g45h5kj5-dirty` as far as I could see.